### ISSUE TITLE
invalidation of empty trigger list is not backward compatible

### ISSUE THREAD

**Issue by joebowbeer** [kedacore/keda#6265]
The recent rejection of empty trigger lists is not backward compatible:

https://github.com/kedacore/keda/pull/5524

See https://github.com/kedacore/keda/issues/5520#issuecomment-2432950971_

KEDA has supported empty trigger lists for years.

The 80% CPU utilization trigger is added implicitly. That is how HPA works, as described in KEDA documentation:

https://keda.sh/docs/2.15/reference/glossary/#hpa

> By default, scales based on CPU/memory usage. KEDA uses HPA to scale Kubernetes clusters and deployments.

We have been relying on it.

I think the best of both worlds solution would be to make this validator optional.

**Comment by zroubalik** [comment]
my 2c: I think that the original behaviour was a bug not a feature (empty triggers list). We never intended this TBH. I think that adding an option to disable this specific validation makes sense, because I understand your reasons.

**Comment by SpiritZhou** [comment]
As an empty trigger configuration will be blocked, it is not reasonable for an empty trigger list to be exempted from this blocking. If an option is added, should it disable the validation for both empty trigger values and empty trigger lists? @zroubalik

**Comment by joebowbeer** [comment]
It's only the invalidation of empty lists that I feel is a breaking change, and should at least be opt-out going forward.

An empty trigger list is not really empty, as two triggers are added implicitly by HPA - whether the list is empty or not, right?

Our "idiom" of relying on an empty trigger list as a base kustomization is convenient because k8s does not support patching of items in a list. For that reason our kustomize base has an empty trigger list, and the kustomize overlays add their specific triggers. (Otherwise, e.g., if we explicitly added CPU and Memory in the base, there would be no way from an overlay to kustomize these.)

**Comment by zroubalik** [comment]
Yeah, empty list - if by empty values is meant a trigger (type) defined without any content.

> Admittedly, this implicit behavior can be confusing. In particular, adding any trigger will remove the implicit cpu trigger...

Yeah, this is one of the reasons I don't like this kind of hidden explicit behaviors :)
